src/pretix/api/views/order.py

```diff
@@ -1905,16 +1908,18 @@ def regenerate(self, request, **kwargs):
             return Response(status=204)
 
     @action(detail=True, methods=['POST'])
+    @transaction.atomic()
     def reissue(self, request, **kwargs):
         inv = self.get_object()
+        order = Order.objects.select_for_update(of=OF_SELF).get(pk=inv.order_id)
```

Comment:
Do we need to already lock the order here? Why not after validation right before we `generate_invoice()` ? Might be micro-optimization though …